[FrameworkBundle] Support use of hyphen in assets package name

damaya opened the pull request:
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no     <!-- see https://symfony.com/bc -->
| Deprecations? | no
| Tests pass?   | yes (Manual tests only)
| Fixed tickets | #28122
| License       | MIT
| Doc PR        | n/a

According to issue symfony/symfony-docs#10442, we tested in a demo bundle, for example in src/AppBundle/Resources/config/config.yml a package using hyphens: app-client-frontend, and withouth the patch it fails because the package is not recognized. With the patch, it works as expected.

```yaml
framework:
    assets:
        packages:
            app-client-frontend:
                version: "%env(FRONTEND_VERSION)%"
                version_format: '%%2$s/dist/%%1$s'
                base_urls:
                  - "%env(FRONTEND_URL)%"
```

damaya committed:
Update Configuration.php

Support for using hyphens on yaml config for packages names, according to issue https://github.com/symfony/symfony/issues/28122#issuecomment-410339787

nicolas-grekas commented on src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php:
```diff
@@ -631,7 +631,7 @@ private function addAssetsSection(ArrayNodeDefinition $rootNode)
                     ->fixXmlConfig('package')
                     ->children()
-                        ->arrayNode('packages')
+                        ->arrayNode('packages')->normalizeKeys(false)
```
should be put on a separate line

nicolas-grekas reviewed (changes requested):
can you please enhance the commit message, and update the PR description according to the required template (filled in)?
see https://github.com/symfony/symfony/blob/master/.github/PULL_REQUEST_TEMPLATE.md

fabpot commented:
Some test should be added as well (example from the related issue would work well here).

damaya commented:
Sure, sorry the delay in my answer, I'll update the PR with required data.

damaya committed:
#28122 Support use of hyphen in asset package name

| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no     <!-- see https://symfony.com/bc -->
| Deprecations? | no
| Tests pass?   | yes (Manual tests only)
| Fixed tickets | #28122
| License       | MIT
| Doc PR        | n/a

According to issue https://github.com/symfony/symfony-docs/pull/10442, we tested in a demo bundle, for example in src/AppBundle/Resources/config/config.yml a package using hyphens: app-client-frontend, and withouth the patch it fails because the package is not recognized. With the patch, it works as expected.
```
framework:
    assets:
        packages:
            app-client-frontend:
                version: "%env(FRONTEND_VERSION)%"
                version_format: '%%2$s/dist/%%1$s'
                base_urls:
                  - "%env(FRONTEND_URL)%"
```

damaya reviewed (commented):
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no     <!-- see https://symfony.com/bc -->
| Deprecations? | no
| Tests pass?   | yes (Manual tests only)
| Fixed tickets | #28122
| License       | MIT
| Doc PR        | n/a

According to issue symfony/symfony-docs#10442, we tested in a demo bundle, for example in src/AppBundle/Resources/config/config.yml a package using hyphens: app-client-frontend, and withouth the patch it fails because the package is not recognized. With the patch, it works as expected.

```yaml
framework:
    assets:
        packages:
            app-client-frontend:
                version: "%env(FRONTEND_VERSION)%"
                version_format: '%%2$s/dist/%%1$s'
                base_urls:
                  - "%env(FRONTEND_URL)%"
```

damaya commented:
Hi there, I updated my commit message and use a new line to improve the syntax. I just did manual tests and this patch worked as expected.
Thanks in advance.

fabpot commented:
@damaya I still think that we need to add a test to avoid any regressions in the future. Can you do that or do you need help?

damaya commented:
> @damaya I still think that we need to add a test to avoid any regressions in the future. Can you do that or do you need help?

I would love to do that, could you please give me a reference for how to implement tests for this according to the standard? thanks so much.

XuruDragon commented:
The continuity of this PR has been achieved here : #30007

nicolas-grekas commented:
Continued in #30007, thank you @damaya
